docs/reference/token-types.md

```diff
@@ -0,0 +1,168 @@
+---
+title: Types of Auth Tokens
+---
+
+Last updated: `December 04th, 2019`
+
+Firefox Accounts uses a number of different types of "token" to keep track of whether a user has been authenticated,
+and of what actions a client is authorized to perform. This document provides an overview of the different types of
+token, their properties and intended uses.
+
+The most important types of token for FxA Replying Parties to know about are:
+
+* [Session Tokens](#session-tokens), long-lived tokens that represent an authenticated user session with the FxA servers.
+* [OAuth Refresh Tokens](#oauth-refresh-tokens), long-lived tokens indicating that the user has authorized a particular client to perform
+  particular actions on their behalf.
+* [OAuth Access Tokens](#oauth-access-tokens), short-lived tokens used by authorized clients when performing particular actions
+  on behalf of the user.
+* [OIDC Identity Tokens](#oidc-identity-tokens), short-lived JWTs used to communicate an authentication event from FxA
+  to a relying party.
+
+FxA has additional types of token that it uses for its own account-management purposes, but which should
+never be visible to Relying Parties:
+
+* [KeyFetch Tokens](#keyfetch-tokens)
+* [Password Change Tokens](#password-change-tokens)
+* [Password Forgot Tokens](#password-forgot-tokens)
+* [Account Reset Tokens](#account-reset-tokens)
+
+Relying Parties that wish to integrate with Firefox Sync may also need to know about these additional types
+of token:
+
+* [BrowserID Assertions](#browserid-assertions)
+* [Sync Storage Credentials](#sync-storage-credentials)
+
+# Token Usage Mechanisms
+
+A ***Bearer Token*** is an opaque string that can be used by any client in posession of the token, simply by presenting
+it verbatim in the request to the server (e.g. in the `Authorization` header). These are the simplest types of token
+for clients to use, but allow for interception and replay attacks. In particular, the server that receives a Bearer Token
+is capable of extracting it and using it to perform other API calls.
+
+A ***Hawk Token*** consists of an opaque string id and corresponding secret key. To use the token, clients must use the
+secret key to sign outgoing requests using [the scheme described here](https://hapi.dev/family/hawk/). This is more
```

Comment:
Hawk should be linked to https://github.com/mozilla/hawk/blob/main/API.md or https://github.com/mozilla/hawk